### DISCUSSION TITLE
Generating new lyn jumps

### DISCUSSION THREAD

**Discussion by JesterWizard** [MokhaLeee/fe8u-cskillsys-kernel#328] [General]
How do I do this? 

In the case of the arena, I figured it was just a case of adding a line to the MiscSkillEffects.event file, but I get an error in WSL telling me there's no rule to make the target?

**Comment by MokhaLeee** [comment]
refer to this https://github.com/MokhaLeee/fe8u-cskillsys-kernel/commit/8133eec7dfaee90d73e96d174d636ab6bf5405bf

**Reply by JesterWizard** [reply]
That's what I was using as my reference. I added the lyn jump line in the Arena C file as well as MiscSkillEffects.event. 

```
[make[1]: *** No rule to make target '/home/username/C-SkillSystem-Mokha/Wizardry/Core/SkillSys/MiscSkillEffects/ArenaSkills/LynJump.event', needed by '/home/username/C-SkillSystem-Mokha/fe8-kernel-dev.gba'.  Stop.
make[1]: *** Waiting for unfinished jobs....
make: *** [makefile:29: all] Error 2]

```

![image](https://github.com/user-attachments/assets/cc7b125f-5a02-44af-a645-380307f80000)
![image](https://github.com/user-attachments/assets/f2ecd95d-1247-4592-8e69-594c6fae13ac)

**Comment by MokhaLeee** [comment]
How did you write in LynJump.event? And you need to put more log

**Reply by JesterWizard** [reply]
I didn't write anything in the LynJump, I had assumed the contents were automatically generated like previous lyn event files?

There's no more log to give you, that's the extent of the error. Those lines are the very last thing in the terminal before the compilation terminates.

**Reply by MokhaLeee** [reply]
Lyn can autumatically generate *.lyn.event, but the LynJump.event need you copy the content by yourself. You need to copy the lyn-jump part to the LynJump.event file.

The purpose to write an additional LynJump.event file is based on one assumption: function names in C-Lib is not fixed, so hackers need to do some preventive programming based on the assumption that an auto-generated LynJump will fail in the future.

**Reply by MokhaLeee** [reply]
If function `ArenaGenerateBaseWeapons` is renamed to such as `ArenaEquipeWeapon` or other names in decomp in the future, you need to find a way to make your hack still work

**Reply by JesterWizard** [reply]
Okay that's fine, at least now I know. 

Using ShopSkill as a basis then, 
```

PUSH
ORG $b5220
ALIGN 4
WORD $46C04778 $E59FC000 $E12FFF1C
POIN GetItemPurchasePrice
POP
```
Presumably the ORG refers to the function address location in the decomp, what about the three WORD values? How are they considered?

**Reply by MokhaLeee** [reply]
Trace in NO$GBA

**Reply by MokhaLeee** [reply]
If you're interested in how this works, you should check out the lyn doc from stanH, https://feuniverse.us/t/ea-asm-tool-lyn-elf2ea-if-you-will/2986/2

**Reply by JesterWizard** [reply]
Thanks! I'll read up on this 👍

**Reply by JesterWizard** [reply]
Okay I got ``LYN_REPLACE_CHECK`` to work with ``ArenaGenerateBaseWeapons`` but I had to place it after the function definition otherwise I got an error from the lyn jump telling me the function was undeclared. 

What confuses me is that in every case I've seen you use it, you place the lyn check above the function definition, what am I missing?

**Reply by MokhaLeee** [reply]
maybe that function declaration is lost in .h file in decomp.

**Reply by JesterWizard** [reply]
So it is. 

![image](https://github.com/user-attachments/assets/4670b15d-ce4f-4ab8-a7b7-bc208a9f2ec3)

**Reply by MokhaLeee** [reply]
You can put the function declarations to *vanilla.h* temporarily. These is the issue on decomp and need be fixed later
